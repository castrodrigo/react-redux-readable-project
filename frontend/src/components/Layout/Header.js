import React from "react";
import styled from "styled-components";
import { Link } from "react-router-dom";

const Nav = styled.nav`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  color: #ffffff;
`;

const LinkFormat = styled(Link)`
  color: #ffffff;
`;

const LinkButton = styled(LinkFormat)`
  border: 1px solid #fff9fb;
  padding: 10px;
`;

const Header = () => (
  <Nav>
    <LinkFormat to={"/"}>
      <h1>Readable</h1>
    </LinkFormat>
    <LinkButton to={"/new"}>New Post</LinkButton>
  </Nav>
);

export default Header;
