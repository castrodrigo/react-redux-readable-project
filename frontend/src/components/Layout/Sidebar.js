import styled from "styled-components";

const Sidebar = styled.div`
  color: #ffffff;
  & ul {
    padding-left: 0.5em;
  }
  & ul li {
    margin-bottom: 0.5em;
  }
  & a {
    font-size: 14px;
    color: #ffffff;
  }
`;

export default Sidebar;
