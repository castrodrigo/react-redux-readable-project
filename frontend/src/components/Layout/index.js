import React from "react";
import styled from "styled-components";
import Header from "./Header";
import Footer from "./Footer";
import Sidebar from "./Sidebar";

const Wrapper = styled.div`
  display: grid;
  grid-template-areas:
    "header header header header header header"
    ". main main main right ."
    "footer  footer footer footer footer footer";
  grid-template-columns: 2em 1fr 1fr 1fr 1fr 2em;
  font-family: Roboto;
`;

const HeaderWrapper = styled.section`
  grid-area: header;
  background: #011936;
  & > * {
    max-width: calc(100vw - 4em);
    margin: 0 auto;
  }
`;

const Content = styled.section`
  grid-area: main;
  padding: 0 0.5em 0.5em 0;
  min-height: calc(100vh - 126px);
`;

const SidebarWrapper = styled.section`
  grid-area: right;
  background: #840032;
  padding: 0 0.75em 0.75em;
  color: #ffffff;
`;

const FooterWrapper = styled.section`
  grid-area: footer;
  background: #002642;
  height: 60px;
`;

const Layout = ({ content, sidebar }) => (
  <Wrapper>
    <HeaderWrapper>
      <Header />
    </HeaderWrapper>
    <Content>{content}</Content>
    <SidebarWrapper>
      <Sidebar>{sidebar}</Sidebar>
    </SidebarWrapper>
    <FooterWrapper>
      <Footer />
    </FooterWrapper>
  </Wrapper>
);

export default Layout;
