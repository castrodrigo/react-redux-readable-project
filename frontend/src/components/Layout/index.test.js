import React from "react";
import { shallow } from "enzyme";
import Layout from "./";
import Sidebar from "./Sidebar";

const sidebar = (
  <ul>
    <li>Item</li>
  </ul>
);

const content = <div>Content</div>;

describe("Layout Render", () => {
  it("Should render correctly", () => {
    const wrapper = shallow(<Layout content={content} sidebar={sidebar} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("Should render sidebar if receives data", () => {
    const wrapper = shallow(<Layout content={content} sidebar={sidebar} />);

    const sidebarCopmonent = wrapper.find("SidebarWrapper");

    expect(sidebarCopmonent.length).toBe(1);
  });

  it("Should not render sidebar if does not receive data", () => {
    const wrapper = shallow(<Layout content={content} />);

    const sidebarCopmonent = wrapper.find("SidebarWrapper");

    expect(sidebarCopmonent.length).toBe(0);
  });

  describe("Data Loading", () => {
    it("Should match content", () => {
      const wrapper = shallow(<Layout content={content} />);

      const contentCopmonent = wrapper.find("Content");

      expect(contentCopmonent.props().children).toEqual(content);
    });
  });

  it("Should match sidebar", () => {
    const wrapper = shallow(<Layout content={content} sidebar={sidebar} />);

    const sidebarCopmonent = wrapper.find(Sidebar);

    expect(sidebarCopmonent.props().children).toEqual(sidebar);
  });
});
