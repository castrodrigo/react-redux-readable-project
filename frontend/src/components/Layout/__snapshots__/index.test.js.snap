// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Render Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Layout
    content={
      <div>
        Content
      </div>
    }
    sidebar={
      <ul>
        <li>
          Item
        </li>
      </ul>
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          <Header />
        </ForwardRef>,
        <ForwardRef>
          <div>
            Content
          </div>
        </ForwardRef>,
        <ForwardRef>
          <ForwardRef>
            <ul>
              <li>
                Item
              </li>
            </ul>
          </ForwardRef>
        </ForwardRef>,
        <ForwardRef>
          <Footer />
        </ForwardRef>,
      ],
      "sidebar": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Header />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": true,
            "rules": Array [
              "
  grid-area: header;
  background: #011936;
  & > * {
    max-width: calc(100vw - 4em);
    margin: 0 auto;
  }
",
            ],
          },
          "displayName": "styled.section",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div>
            Content
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Content",
          },
          "ref": null,
          "rendered": "Content",
          "type": "div",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bZQynM",
            "isStatic": true,
            "rules": Array [
              "
  grid-area: main;
  padding: 0 0.75em 0.75em 0;
  min-height: calc(100vh - 146px);
  padding-top: 1em;
",
            ],
          },
          "displayName": "Content",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bZQynM",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef>
            <ul>
              <li>
                Item
              </li>
            </ul>
          </ForwardRef>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ul>
              <li>
                Item
              </li>
            </ul>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <li>
                Item
              </li>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Item",
              },
              "ref": null,
              "rendered": "Item",
              "type": "li",
            },
            "type": "ul",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": true,
              "rules": Array [
                "
  color: #ffffff;
  & ul {
    padding-left: 0.5em;
  }
  & ul li {
    margin-bottom: 0.5em;
  }
  & a {
    font-size: 14px;
    color: #ffffff;
  }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gzVnrw",
            "isStatic": true,
            "rules": Array [
              "
  grid-area: right;
  background: #840032;
  padding: 1em 0.75em 0.75em;
",
            ],
          },
          "displayName": "SidebarWrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-gzVnrw",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Footer />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htoDjs",
            "isStatic": true,
            "rules": Array [
              "
  grid-area: footer;
  background: #002642;
  height: 60px;
",
            ],
          },
          "displayName": "styled.section",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htoDjs",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-ifAKCX",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-template-areas:
    \\"header header header header header header\\"
    \\"",
          [Function],
          "\\"
    \\"footer footer footer footer footer footer\\";
  grid-template-columns: 2em 1fr 1fr 1fr 1fr 2em;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-ifAKCX",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <Header />
          </ForwardRef>,
          <ForwardRef>
            <div>
              Content
            </div>
          </ForwardRef>,
          <ForwardRef>
            <ForwardRef>
              <ul>
                <li>
                  Item
                </li>
              </ul>
            </ForwardRef>
          </ForwardRef>,
          <ForwardRef>
            <Footer />
          </ForwardRef>,
        ],
        "sidebar": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Header />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-EHOje",
              "isStatic": true,
              "rules": Array [
                "
  grid-area: header;
  background: #011936;
  & > * {
    max-width: calc(100vw - 4em);
    margin: 0 auto;
  }
",
              ],
            },
            "displayName": "styled.section",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-EHOje",
            "target": "section",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div>
              Content
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Content",
            },
            "ref": null,
            "rendered": "Content",
            "type": "div",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bZQynM",
              "isStatic": true,
              "rules": Array [
                "
  grid-area: main;
  padding: 0 0.75em 0.75em 0;
  min-height: calc(100vh - 146px);
  padding-top: 1em;
",
              ],
            },
            "displayName": "Content",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bZQynM",
            "target": "section",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef>
              <ul>
                <li>
                  Item
                </li>
              </ul>
            </ForwardRef>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <ul>
                <li>
                  Item
                </li>
              </ul>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <li>
                  Item
                </li>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Item",
                },
                "ref": null,
                "rendered": "Item",
                "type": "li",
              },
              "type": "ul",
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
  color: #ffffff;
  & ul {
    padding-left: 0.5em;
  }
  & ul li {
    margin-bottom: 0.5em;
  }
  & a {
    font-size: 14px;
    color: #ffffff;
  }
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-gzVnrw",
              "isStatic": true,
              "rules": Array [
                "
  grid-area: right;
  background: #840032;
  padding: 1em 0.75em 0.75em;
",
              ],
            },
            "displayName": "SidebarWrapper",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-gzVnrw",
            "target": "section",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Footer />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htoDjs",
              "isStatic": true,
              "rules": Array [
                "
  grid-area: footer;
  background: #002642;
  height: 60px;
",
              ],
            },
            "displayName": "styled.section",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htoDjs",
            "target": "section",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-ifAKCX",
          "isStatic": false,
          "rules": Array [
            "
  display: grid;
  grid-template-areas:
    \\"header header header header header header\\"
    \\"",
            [Function],
            "\\"
    \\"footer footer footer footer footer footer\\";
  grid-template-columns: 2em 1fr 1fr 1fr 1fr 2em;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-ifAKCX",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
